Unity-Technologies/ScriptableRenderPipeline
# Universal shadowmask surport

### Checklist for PR maker
- Have you added a Label? : No
- Have you added a label for backport (if needed)? : No
- Have you added a changelog? Add a line in changelog in URP
- Have you updated or added the documentation for you PR? When property name is changed, when a feature behavior is change, when adding a new features, think to update the documentation in the same PR. No
- Have you added a graphic test for your PR (if needed)? When adding new feature or discovering a bug that isn't cover by a test, please add a graphic test.No

---
### Purpose of this PR
Shadowmask feature is in research in URP. My PR support shadowmask mode in URP.
---
### Testing status

![multi shaodow](https://github.com/sienaiwun/publicImgs/blob/master/imgs/multi_light.png?raw=true)

The shadow information is baked in the shadowmask image. Even the workshop sets on the floor are removed, we can still see the visibility to the directional lights as well as the other two colored point lights. 

![baked shaodow](https://github.com/sienaiwun/publicImgs/blob/master/imgs/shadowmask.png?raw=true)

**Manual Tests**: What did you do?
1. Create the default sample scene.
2. Add 3 points lights.
3. Change the directional light and 3 points lights' “mode” to "mix".
4. Set the whole workshop sets as "static" 
5. Change pipeline setting's main light and additional lights' "cast shadow" to "Off: 
6. In Rendering/Lighting Settings, set "Compress Lightmaps" as "Off" because of another [issue](https://issuetracker.unity3d.com/issues/selected-texture-format-argb-16-bit-is-not-valid-error-is-shown-after-manually-baking-shadowmask), mixed lighting mode as "shadowmask"
7. Turn off "auto Generate", Bake the light, generate lighting. 
8. Move the working set, see the visibility info is baked in shadowmask
